events/event_types/event.py

```diff
@@ -7,3 +10,10 @@ def __init__(self, time):
             time (int):   The time at which the event occurs.
         """
         self.time = time
+
+    @abc.abstractmethod
+    def execute(self):
```

Comment:
Hmmm... do we need the *events* to do the execution? I feel like either components handle events (e.g. a component sees an event and then it has its own selection of what to do with it) or the main network class handles events (e.g. a packet was sent from A to B via link L, so tell link L to send the packet).